spack/packages/dla-future/package.py

```diff
@@ -136,18 +136,51 @@ def cmake_args(self):
         args.append(self.define_from_variant("BUILD_SHARED_LIBS", "shared"))
 
         # BLAS/LAPACK
-        if self.spec["lapack"].name in INTEL_MATH_LIBRARIES:
+        if spec["lapack"].name in INTEL_MATH_LIBRARIES:
+            mkl_provider = spec["lapack"].name
+
             vmap = {
-                "none": "seq",
-                "openmp": "omp",
-                "tbb": "tbb",
-            }  # Map MKL variants to LAPACK target name
-            mkl_threads = vmap[spec["intel-mkl"].variants["threads"].value]
-            # TODO: Generalise for intel-oneapi-mkl
-            args += [
-                self.define("DLAF_WITH_MKL", True),
-                self.define("MKL_LAPACK_TARGET", f"mkl::mkl_intel_32bit_{mkl_threads}_dyn"),
-            ]
+                    "intel-oneapi-mkl": {
+                        "threading": {
+                            "none": "sequential",
+                            "openmp": "gnu_thread",
+                            "tbb": "tbb_thread",
+                            },
+                        "mpi": {
+                            "mpich": "intelmpi",
+                            "openmpi": "openmpi",
+                            }
+                        },
+                    "intel-mkl": {
+                        "threading": {
+                            "none": "seq",
+                            "openmp": "omp",
+                            "tbb": "tbb",
+                            },
+                        "mpi": {
+                            "mpich": "mpich",
+                            "openmpi": "ompi",
+                            }
+                        },
+                    }
+
+            if mkl_provider not in vmap.keys():
+                raise RuntimeError("{0} is not supported".format(mkl_provider))
```

Comment:
If I remember correctly, these messages are usually printed without much context, so do you think it's worth changing it to something more verbose like:
```suggestion
                raise RuntimeError("DLA-Future does not support {0} as the lapack provider".format(mkl_provider))
```
?